packages/consoledot-api/src/instancePermission.ts

```diff
@@ -0,0 +1,26 @@
+import type { KafkaInstanceEnhanced } from "consoledot-api/src/transformers/kafkaRequestToKafkaInstanceEnhanched";
+import { SuspendedStatuses } from "ui-models/src/models/kafka";
+import type { Status } from "ui-models/src/models/kafka";
+
+export function canOpenConnection(instance: KafkaInstanceEnhanced): boolean {
+  return instance ? !SuspendedStatuses.includes(instance.status) : false;
```

Comment:
Shouldn't this logic include only the ready states? I can't connect to a failed or deleted instance, right?

Reply:
@riccardo-forina , I checked the old code, connection action in the kebab menu will be disable when the instance status is suspended

Reply:
I'm not sure we should assume the old code was right. Actually, I'd urge you to challenge it all the time :)

Let's reason about this: what good does having the URL to connect to an instance that's being deleted? Or that's failed? Isn't the only safe moment to connect to the instance when the instance is ready?

Reply:
When we are deleting the instance, kebab actions will not be visible

Reply:
That doesn't mean the logic can be flawed.

Reply:
@riccardo-forina I reverted the logic to include only the ready state, But the only concern I have is the connection option will be disabled, do we need to show some message about why the option is disabled, and also the user can still access the connection tab by click on the details option will that be a good option?

Reply:
If the option in the kebab menu is disabled, the tab should be disabled as well. A tooltip to explain why the tab/action is disabled is a good idea but maybe let's do it in a future PR